AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/physx_collider.py

```diff
@@ -0,0 +1,318 @@
+
+        Used to toggle the Physx Collider's In Scene Queries Property.
+        """
+        self.component.toggle_property_switch(self.Path.IN_SCENE_QUERIES)
+
+    def set_offset(self, x: float, y: float, z: float) -> None:
+        """
+        Property Type, Default Visibility - ('Vector3', 'Visible')
+
+        Used to set the Physx Collider's Offset Property.
+        """
+        self.component.set_component_property_value(self.Path.OFFSET, math.Vector3(float(x), float(y), float(z)))
+
+    def set_rotation(self, x: float, y: float, z: float) -> None:
+        """
+        Property Type, Default Visibility - ('Quaternion', 'Visible')
+
+        Used to set the Physx Collider's Rotation Property.
+        """
+        rotation = math.Quaternion()
+        rotation.SetFromEulerDegrees(math.Vector3(float(x), float(y), float(z)))
+        self.component.set_component_property_value(self.Path.ROTATION, rotation)
+
+    def set_tag(self, tag: chr) -> None:
+        """
+        Property Type, Default Visibility - ('AZStd::string', 'Visible')
+
+        Used to set the Physx Collider's Tag Property.
+        """
+        # TODO: GHI #12634 - For some reason I can't get this to work. Says it takes an AZStd::string, but it won't take
+        #  one or a character.
+        assert NotImplementedError
+        self.component.set_component_property_value(self.Path.TAG, tag)
+
+    def set_rest_offset(self, offset: float) -> None:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to set the Physx Collider's Reset Offset Property.
+        """
+        self.component.set_component_property_value(self.Path.RESET_OFFSET, offset)
+
+    def set_contact_offset(self, offset: float) -> None:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to set the Physx Collider's Contact Offset Property.
+        """
+        self.component.set_component_property_value(self.Path.CONTACT_OFFSET, offset)
+
+    def toggle_draw_collider(self) -> None:
+        """
+        Property Type, Default Visibility - ('bool', 'Visible')
+
+        Used to toggle the Physx Collider's Draw Collider Property.
+        """
+        self.component.toggle_property_switch(self.Path.DRAW_COLLIDER)
+
+    def set_physx_material_from_path(self, asset_product_path: str) -> None:
+        """
+        Property Type, Default Visibility - ('Asset<Physics::MaterialAsset>', 'Visible')
+
+        Used to set the Physx Collider's PhysX Material from a user provided asset product path. This will return an
+            error if a source path is provided.
+        """
+        # GHI #12503 PhysX Collider Component's Physic Material field(s) return unintuitive property tree paths.
+        assert NotImplementedError
+        assert self.component.is_property_visible(self.Path.PHYSX_MATERIAL_ASSET), \
+            f"Failure: Cannot set Physx Material when property is not visible."
```

Comment:
As this is replicating the intended usage, I decided that checking the visibility was a valid use case. This also guards us from a suddenly failing test if somebody ever decided to go, "Hey, let's not allow a non-visible property from being set by the bus because that caused issues over here."

Reply:
refactored